examples/nlp/md/text_generation_with_miniature_gpt.md

```diff
@@ -0,0 +1,668 @@
+        while num_tokens_generated <= self.max_tokens:
+            pad_len = maxlen - len(start_tokens)
+            sample_index = len(start_tokens) - 1
+            if pad_len < 0:
+                x = start_tokens[:maxlen]
+                sample_index = maxlen - 1
+            elif pad_len > 0:
+                x = start_tokens + [0] * pad_len
+            else:
+                x = start_tokens
+            x = np.array([x])
+            y, _ = self.model.predict(x)
+            sample_token = self.sample_from(y[0][sample_index])
+            tokens_generated.append(sample_token)
+            start_tokens.append(sample_token)
+            num_tokens_generated = len(tokens_generated)
+        txt = " ".join(
+            [self.detokenize(_) for _ in self.start_tokens + tokens_generated]
+        )
+        print(f"generated text:\n{txt}\n")
+
+
+# Tokenize starting prompt
+word_to_index = {}
+for index, word in enumerate(vocab):
+    word_to_index[word] = index
+
+start_prompt = "this movie is"
+start_tokens = [word_to_index.get(_, 1) for _ in start_prompt.split()]
+num_tokens_generated = 10
+text_gen_callback = TextGenerator(num_tokens_generated, start_tokens, vocab)
+
+
+```
+
+---
+## Train
+
+Note: This code should preferably be run on GPU.
+
+
+
+```python
+model = create_model()
+
+model.fit(text_ds, verbose=2, epochs=30, callbacks=[text_gen_callback])
+
+```
+
+<div class="k-default-codeblock">
+```
+Epoch 1/30
+generated text:
+this movie is a movie a great idea of the movie ive ever seen
+```
+</div>
+    
+<div class="k-default-codeblock">
+```
+1563/1563 - 17s - loss: 5.7170 - dense_6_loss: 5.7170
+Epoch 2/30
+generated text:
+this movie is so bad i was a big fan of the worst movie
+```
+</div>
+    
+<div class="k-default-codeblock">
+```
+1563/1563 - 17s - loss: 4.9930 - dense_6_loss: 4.9930
+Epoch 3/30
+generated text:
+this movie is terrible it is that its the worst movie ever made me
+```
+</div>
+    
+<div class="k-default-codeblock">
+```
+1563/1563 - 17s - loss: 4.7348 - dense_6_loss: 4.7348
+Epoch 4/30
+generated text:
+this movie is one of those films that makes me wonder that this has
+```
+</div>
+    
+<div class="k-default-codeblock">
+```
+1563/1563 - 16s - loss: 4.5729 - dense_6_loss: 4.5729
+Epoch 5/30
+generated text:
+this movie is so bad it could have been made for a movie with
+```
+</div>
+    
+<div class="k-default-codeblock">
+```
+1563/1563 - 16s - loss: 4.4569 - dense_6_loss: 4.4569
+Epoch 6/30
+generated text:
+this movie is so awful i think that the [UNK] could have seen a
+```
+</div>
+    
+<div class="k-default-codeblock">
+```
+1563/1563 - 18s - loss: 4.3682 - dense_6_loss: 4.3682
+Epoch 7/30
+generated text:
+this movie is just a [UNK] movie the acting was a bit of the
+```
+</div>
+    
+<div class="k-default-codeblock">
+```
+1563/1563 - 16s - loss: 4.2981 - dense_6_loss: 4.2981
+Epoch 8/30
+generated text:
+this movie is one of the worst movies i have ever seenbr br it
```

Comment:
In general with raw IMDB data you should strip the `<br>` tags (like we do in the text classification example). Otherwise only the `<` and `>` characters get stripped and you end up with random "br" strings in your text